compiler/rustc_parse/src/parser/item.rs

```diff
@@ -176,7 +188,7 @@ impl<'a> Parser<'a> {
         }
 
         // At this point, we have failed to parse an item.
-        if !matches!(vis.kind, VisibilityKind::Inherited) {
+        if !matches!(vis.kind, VisibilityKind::Inherited) && !self.token.is_keyword(kw::Let) {
```

Comment:
Can you explain why you added this? It seems risky to be doing this.

What happens when someone types `pub(crate) let x = 1;` in a module?

Reply:
You are right